dGame/dUtilities/SlashCommandHandler.cpp

```diff
@@ -1278,22 +1278,22 @@ void SlashCommandHandler::HandleChatCommand(const std::u16string& command, Entit
 	}
 
 	if ((chatCommand == "giveuscore") && args.size() >= 1 && entity->GetGMLevel() >= eGameMasterLevel::DEVELOPER) {
-		int32_t uscore;
-
-		if (!GeneralUtils::TryParse(args[0], uscore)) {
+		const auto uscoreOptional = GeneralUtils::TryParse<int32_t>(args[0]);
+		if (!uscoreOptional) {
 			ChatPackets::SendSystemMessage(sysAddr, u"Invalid uscore.");
 			return;
 		}
+		const int32_t uscore = uscoreOptional.value();
 
 		CharacterComponent* character = entity->GetComponent<CharacterComponent>();
 		if (character) character->SetUScore(character->GetUScore() + uscore);
 		// MODERATION should work but it doesn't.  Relog to see uscore changes
 
 		eLootSourceType lootType = eLootSourceType::MODERATION;
 
-		int32_t type;
-		if (args.size() >= 2 && GeneralUtils::TryParse(args[1], type)) {
-			lootType = static_cast<eLootSourceType>(type);
+		const auto type = GeneralUtils::TryParse<eLootSourceType>(args[1]);
+		if (args.size() >= 2 && type) {
+			lootType = type.value();
```

Comment:
The index check here was moved to be after the value is indexed

Reply:
Same issue as the other one. If I fold that into the if statement, I'm not sure how to get the value out without calling the function a second time.

Reply:
Figured it out.